data_structures/src/chain.rs

```diff
@@ -2099,6 +2101,137 @@ impl UnspentOutputsPool {
     }
 }
 
+/// List of unspent outputs that can be spent by this node
+#[derive(Clone, Debug, Default, Eq, PartialEq, Serialize, Deserialize)]
+pub struct OwnUnspentOutputsPool {
+    /// Map of output pointer to timestamp
+    /// Those UTXOs have a timestamp value to avoid double spending
+    map: HashMap<OutputPointer, u64>,
+    /// Counter of UTXOs bigger than the minimum collateral
+    bigger_than_min_counter: usize,
+}
+
+impl OwnUnspentOutputsPool {
+    pub fn get<Q: ?Sized>(&self, k: &Q) -> Option<&u64>
+    where
+        OutputPointer: std::borrow::Borrow<Q>,
+        Q: std::hash::Hash + Eq,
+    {
+        self.map.get(k)
+    }
+
+    pub fn get_mut<Q: ?Sized>(&mut self, k: &Q) -> Option<&mut u64>
+    where
+        OutputPointer: std::borrow::Borrow<Q>,
+        Q: std::hash::Hash + Eq,
+    {
+        self.map.get_mut(k)
+    }
+
+    pub fn contains_key<Q: ?Sized>(&self, k: &Q) -> bool
+    where
+        OutputPointer: std::borrow::Borrow<Q>,
+        Q: std::hash::Hash + Eq,
+    {
+        self.map.contains_key(k)
+    }
+
+    pub fn insert(&mut self, k: OutputPointer, v: u64, bigger_than_min: bool) -> Option<u64> {
+        let old_ts = self.map.insert(k, v);
+        if bigger_than_min {
```

Comment:
Here you can also add a check like the one in `remove`

```suggestion
        if old_ts.is_none() && bigger_than_min {
```